compiler/ilgen/IlBuilder.cpp

```diff
@@ -919,12 +919,22 @@ TR::IlValue *
 IlBuilder::ConvertTo(TR::IlType *t, TR::IlValue *v)
    {
    appendBlock();
-   TR::IlValue *convertedValue = newValue(t);
    TR::DataType t1 = v->getSymbol()->getDataType();
    TR::DataType t2 = t->getPrimitiveType();
+   if (t1 == t2)
+      {
+      TraceIL("IlBuilder[ %p ]::%d is ConvertTo %d (already has type %s)\n", this, v->getCPIndex(), v->getCPIndex(), t->getName());
+      return v;
+      }
+
    TR::ILOpCodes convertOp = TR::DataType::getDataTypeConversion(v->getSymbol()->getDataType(), t->getPrimitiveType());
+   TR_ASSERT(convertOp != TR::BadILOp, "unknown type conversion required");
```

Comment:
In the assert message, would it make more sense to say "unknown type conversion __requested__" ? Also, it would be helpful to have some debug information right in the message, eg: source type, target type, `IlBuilder` object.